### PR TITLE
Extract metadata from pom.xml

### PR DESCRIPTION
**Pull request by sridamul**
Initial Iteration to extract essential metadata from pom.xml
### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by jonesbusy** [comment]
To fix spotbug and spotless

File: plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/extractor/PluginMetadata.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package io.jenkins.tools.pluginmodernizer.core.extractor;
+
+import java.util.List;
+import java.util.Map;
+
+import org.openrewrite.maven.tree.*;
+
+public class PluginMetadata {
```
Comment (jonesbusy) [inline plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/extractor/PluginMetadata.java:8]: I would keep it very very simple for now.

Probably a first iteration would be

pluginName
parentVersion
jenkinsVersion

What do you think ?

The rest could be added later if we need to

Comment (sridamul) [reply plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/extractor/PluginMetadata.java:8]:
```    
    private String pluginName;
    private boolean isLicensed;
    private boolean hasDevelopersTag;
    private boolean hasJavaLevel;
    private boolean usesHttps;
    private List<Dependency> dependencies;
    private String jenkinsVersion;
    private Parent pluginParent;
    private Map<String, String> properties;
```

Already extracted the things in this list. Will commit after resolving spotbugs and checkstyle issues.

**Comment by jonesbusy** [comment]
Thanks, lets merge it as is and improve in the future

**Review by jonesbusy — APPROVED** [review]

**Pull request merged**